handlers/messages.go

```diff
@@ -1,168 +1,127 @@
+// Functions handling messages and message-related things
 package handlers
 
 import (
 	"fmt"
 	"log"
-	"regexp"
-	"strconv"
-	"strings"
 	"time"
 
 	"github.com/bwmarrin/discordgo"
 )
 
-const BOT_PREFIX = "!"
-
-func MessageHandler(session *discordgo.Session, msg *discordgo.MessageCreate) {
-	// if the message is from a bot, return (prevent loops)
-	//  or if the message is only an image, it's content is empty
-	if msg.Author.Bot || msg.Content == "" {
-		return
-	}
-
-	// ping command - check for life
-	if msg.Content == "ping" {
-		log.Println("msghandler: caught ping command")
-		// mention the sender back and say pong!
-		if _, err := session.ChannelMessageSend(msg.ChannelID, msg.Author.Mention()+", pong!"); err != nil {
-			log.Println("\u001b[31mERROR:\u001b[0m", err.Error())
-			return
-		}
-	}
-
-	// if the first character of the message matches our bot prefix
-	if msg.Content[0:1] == BOT_PREFIX {
-		words := strings.Fields(msg.Content) // split message string on space
-		command := words[0][1:]              // get first word and remove BotPrefix
-		args := words[1:]                    // get the rest of the message
-
-		switch command { // switch on command word
-		case "echo": // echo the message that was sent
-			log.Println("msghandler: caught echo command")
-			echo(session, msg, args)
-
-		case "glizzy": // glizzy
-			log.Println("msghandler: caught glizzy command")
-			// TODO: get these emojis not in a hard-coded way
-			_, err := session.ChannelMessageSend(msg.ChannelID, "<a:glizzyR:991176701063221338>"+strings.Join(args, " ")+"<a:glizzyL:991176582402150531>")
-			if err != nil {
-				log.Println("\u001b[31mERROR:\u001b[0m", err.Error())
-				return
-			}
-
-		case "remindme", "remind":
-			log.Println("msghandler: caught remindme command")
-			remindMe(session, msg, args)
-
-		case "shuffle":
-			log.Println("msghandler: caught shuffle command")
-			shuffleVCs(session, msg)
-
-		default: // if the command does not match an existing one, return
-			return
-		}
-	}
+// /ping driver function, responds to verify bot life
+func ping(session *discordgo.Session, interaction *discordgo.InteractionCreate) {
```

Comment:
This pattern is looking much better for breaking apart commands. I would start to look at inverting the control of commands though. Rather than have a handler which lists everything each command should register themselves to the handler. This better isolates each command which makes for easier to maintain code. You can then write a test for each command independently of the others.

Reply:
I'll look into this, thanks